hs/src/Reach/AST/SL.hs

```diff
@@ -607,6 +650,81 @@ data SLPrimitive
   | SLPrim_Token_destroyed
   deriving (Eq, Generic)
 
+instance Equiv SLPrimitive where
+  equiv a b = case (a,b) of
```

Comment:
This case sucks so bad... I don't want to maintain it... I want to ignore it all except for the primop case. I know you worked hard but sometimes when you see the result you learn that it was a bad idea to send the soldiers into that valley

Reply:
lol it's brutal yea. I think it's not so bad to require adding a new case to the Equiv instance for SLPrimitive if we ever need it. Currently it's defined 
```
instance Equiv SLPrimitive where
  equiv a b = case (a,b) of
    (SLPrim_op p1, SLPrim_op p2) -> equiv p1 p2
    _ -> False
```
which seems less bad. I'll make a comment in the code to explain what's going on here

Reply:
Maybe there's a special thing that Generics could do: Could we do something like "If a and b are the same constructor and they have no parameters, then compare them with `==`?"

Reply:
Definitely. We can do a lot in Core with Generics. Once I get this passing I'll try that out. btw I think it would also work with constructors with 1 value, because it could assume `(Something v1, Something v2) -> equiv v1 v2` if we restrict the fields to having `Equiv` instances. In fact, we could probably define something smarter for all cases by just defining how to access the "equivable" fields with a lens or simple accessor.
```
class Equiv a where
  accessors :: [(sum -> f)] -> [(f1,f2)]
  equiv :: sum -> sum -> Bool
  default equiv a b = all (equiv . _) . accessors a <> accessors b
  
instance Equiv Something where
  accessors = [\(Var _ _ v) -> v]
```

Reply:
Let's not get too excited ;)

Reply:
lol for sure. I'd actually be more interested in combining the `NewerVar` and `Equiv` classes into one class, but I'm not really sure if that makes sense semantically. Either way, it's probably fine for now